### PR TITLE
Video output saving

### PR DESCRIPTION
**Pull request by Davidy22**
@Trisanu-007 See IOBase for base class to help with single image saving. Commit to resolve issue #5 and groundwork for #4.

### CODE REVIEW

**Comment by Trisanu-007** [comment]
Will do !

**Comment by Davidy22** [comment]
If you don't see issue with a pull request and it runs, you can approve and merge

File: src/fileIO.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                    Image.composite(char, self.charBack, char),
+                    (x * self.fx, y * self.fy),
+                )
+        return array(out)
+
+
+class videoIO(IOBase):  # TODO: Other video filetypes
+    """ASCII to video saver"""
+
+    def __init__(self, dim: Tuple[int, int], dest: str = "out.avi"):
```
Comment (dhananjaylatkar) [inline src/fileIO.py:68]: By default, we can store video/image data in separate folder with timestamp as a name of file.

File: src/fileIO.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+from cv2 import VideoWriter, VideoWriter_fourcc
+from numpy import array
+from PIL import Image, ImageDraw, ImageFont
+
+
+class IOBase:
+    """ASCII conversion module, subclass for access to convert()"""
+
+    def __init__(self):
+        self.colours = load(open("dat/colours.json"))
```
Comment (dhananjaylatkar) [inline src/fileIO.py:13]: `data` instead of `dat`

File: src/fileIO.py (hunk 1 above)
Comment (Davidy22) [reply src/fileIO.py:68]: File saving utility honestly shouldn't even have a default, and just throw an error like open() when the user neglects to specify, I just had one for convenience

File: src/dat/colours.json [hunk 3]
```diff
@@ -0,0 +1 @@
...{"r": 148, "g": 148, "b": 148}, "hsl": {"h": 0, "s": 0, "l": 58}, "name": "Grey58"}, "247": {"hexString": "#9e9e9e", "rgb": {"r": 158, "g": 158, "b": 158}, "hsl": {"h": 0, "s": 0, "l": 61}, "name": "Grey62"}, "248": {"hexString": "#a8a8a8", "rgb": {"r": 168, "g": 168, "b": 168}, "hsl": {"h": 0, "s": 0, "l": 65}, "name": "Grey66"}, "249": {"hexString": "#b2b2b2", "rgb": {"r": 178, "g": 178, "b": 178}, "hsl": {"h": 0, "s": 0, "l": 69}, "name": "Grey70"}, "250": {"hexString": "#bcbcbc", "rgb": {"r": 188, "g": 188, "b": 188}, "hsl": {"h": 0, "s": 0, "l": 73}, "name": "Grey74"}, "251": {"hexString": "#c6c6c6", "rgb": {"r": 198, "g": 198, "b": 198}, "hsl": {"h": 0, "s": 0, "l": 77}, "name": "Grey78"}, "252": {"hexString": "#d0d0d0", "rgb": {"r": 208, "g": 208, "b": 208}, "hsl": {"h": 0, "s": 0, "l": 81}, "name": "Grey82"}, "253": {"hexString": "#dadada", "rgb": {"r": 218, "g": 218, "b": 218}, "hsl": {"h": 0, "s": 0, "l": 85}, "name": "Grey85"}, "254": {"hexString": "#e4e4e4", "rgb": {"r": 228, "g": 228, "b": 228}, "hsl": {"h": 0, "s": 0, "l": 89}, "name": "Grey89"}, "255": {"hexString": "#eeeeee", "rgb": {"r": 238, "g": 238, "b": 238}, "hsl": {"h": 0, "s": 0, "l": 93}, "name": "Grey93"}}
```
Comment (dhananjaylatkar) [inline src/dat/colours.json:1]: format in human readable ????

Comment (Davidy22) [reply src/dat/colours.json]: It's a constants file ripped from a reference, if a human edits this they're making a mistake, treat like a binary like everything else that'll go in /data

File: src/fileIO.py (hunk 2 above)
Comment (Davidy22) [reply src/fileIO.py]: 9668fdd7d63ecc4ffa86338cb141eb5a3282c034 and 9668fdd7d63ecc4ffa86338cb141eb5a3282c034

**Comment by Davidy22** [comment]
just going to exercise benevolent dictator power and merge since the only objections are to the content of constants and #4 depends on this

**Pull request merged**